templates/app/angular-default/template/.eslintrc.json

```diff
@@ -0,0 +1,104 @@
+/**
+ * -----------------------------------------------------
+ * NOTES ON CONFIGURATION STRUCTURE
+ * -----------------------------------------------------
+ *
+ * Out of the box, ESLint does not support TypeScript or HTML. Naturally those are the two
+ * main file types we care about in Angular projects, so we have to do a little extra work
+ * to configure ESLint exactly how we need to.
+ *
+ * Fortunately, ESLint gives us an "overrides" configuration option which allows us to set
+ * different lint tooling (parser, plugins, rules etc) for different file types, which is
+ * critical because our .ts files require a different parser and different rules to our
+ * .html (and our inline Component) templates.
+ */
+ {
+  "root": true,
+  "overrides": [
+    /**
+     * -----------------------------------------------------
+     * TYPESCRIPT FILES (COMPONENTS, SERVICES ETC) (.ts)
+     * -----------------------------------------------------
+     */
+    {
+      "files": ["*.ts"],
+      "parserOptions": {
+        "project": [
+          "tsconfig.*?.json"
+        ],
+        "createDefaultProgram": true
+      },
+      "extends": ["plugin:@angular-eslint/recommended"],
```

Comment:
should this be top-level, or does this only apply for typescript files (i.e. what rules should apply for JS files - or are they not supported in angular projects?)

Reply:
Generally you don't use JS in an Angular project which is why everything is tailored around TS. I didn't wrote this configuration myself, it is the recommended default config from [angular-eslint](https://github.com/angular-eslint/angular-eslint). I'd say if linting of JS files is required users can always add their own config targeted for `* .js` files.